chain/network-primitives/src/blacklist.rs

```diff
@@ -42,10 +42,13 @@ impl Blacklist {
     }
 
     /// Returns whether given address is on the blacklist.
-    pub fn contains(&self, addr: &std::net::SocketAddr) -> bool {
-        match self.0.get(&addr.ip()) {
+    pub fn contains(&self, addr: impl PeerWithAddress) -> bool {
+        match addr.get_address() {
             None => false,
-            Some(ports) => ports.contains(addr.port()),
```

Comment:
Why do you make the contains() method generic? The get_address thing can always be called explicitly at the call site anyway.